concepts/websockets.textile

```diff
@@ -0,0 +1,593 @@
+
+h4(#minimal-code-samples-4). Minimal code samples
+
+*Client (Browser, before bundling):*
+
+```[js]
+var sock = new SockJS('https://mydomain.com/my_prefix');
+sock.onopen = function() {
+  console.log('open');
+  sock.send('test');
+};
+
+sock.onmessage = function(e) {
+  console.log('message', e.data);
+  sock.close();
+};
+
+sock.onclose = function() {
+  console.log('close');
+};
+```
+
+*Server (Node.js):*
+
+```[js]
+var http = require('http');
+var sockjs = require('sockjs');
+
+var echo = sockjs.createServer({ sockjs_url: 'http://cdn.jsdelivr.net/sockjs/1.0.1/sockjs.min.js' });
+echo.on('connection', function(conn) {
+  conn.on('data', function(message) {
+    conn.write(message);
+  });
+  conn.on('close', function() {});
+});
+
+var server = http.createServer();
+echo.installHandlers(server, {prefix:'/echo'});
+server.listen(9999, '0.0.0.0');
+```
+
+h3(#deepstream-server). deepstream server
+
+As per the "deepstream server website":https://deepstreamhub.com/open-source/, _"deepstream is a powerful websocket server that syncs realtime data between browsers, smartphones, backends and the IoT"_. Unlike the other server-based libraries above, deepstream server is its own server application which must be downloaded and installed on your server. Bindings are provided for "integration with different languages and environments":https://deepstreamhub.com/docs/, such as Node.js and Java. As a full-featured server, it comes with a full suite of capabilities, including pub/sub channels, presence, authentication, and more.
+
+h2(#scaling-beyond-a-single-server). Scaling beyond a single server
+
+The number of concurrent connections a server can handle is rarely the bottleneck when it comes to server load. Most decent WebSocket servers can support hundreds of thousands or even millions of concurrent connections, but what's the workload required to process and respond to messages once the WebSocket server process has handled receipt of the actual data? Typically there will be all kinds of potential concerns, such as reading and writing to and from a database, integration with a game server, allocation and management of resources for each client, and so forth. As soon as one machine is unable to cope with the workload, you'll need to start adding additional servers, which means now you'll need to start thinking about load-balancing, synchronization of messages among clients connected to different servers, generalised access to client state irrespective of connection lifespan or the specific server that the client is connected to - the list goes on and on.
+
+Such concerns are deserving of an article all of their own, and you'll find plenty in "the _Engineering_ section of the Ably blog":https://blog.ably.io - see in particular Matt O'Riordan's writeup on "some of the complexities that were dealt with when building Ably":https://blog.ably.io/the-story-of-how-ably-came-to-be-1d859f2c9a90.
```

Comment:
As mentioned in the comment [here](https://github.com/ably/docs/pull/495#discussion_r213555306), please change the Engineering blog's link to https://blog.ably.io/tagged/realtime-tech

Reply:
'Matt' -> 'Matthew'